ofono/src/sms.c

```diff
@@ -1174,6 +1387,13 @@ static const GDBusMethodTable sms_manager_methods[] = {
 			GDBUS_ARGS({ "to", "s" }, { "text", "s" }),
 			GDBUS_ARGS({ "path", "o" }),
 			sms_send_message) },
+	{ GDBUS_ASYNC_METHOD("SendDataMessage",
+			GDBUS_ARGS(
+				{ "to", "s" },
+				{ "data", "v" }, // s|ay
+				{ "port", "q"}),
```

Comment:
Need to pass both source and destination ports as separate parameters. And let's reserve another 'u' parameter for flags, and 0x01 bit for the "delivery report" flag.

Reply:
Added src and dst ports as well as flags.